laravel-app/app/Policies/UserPostPolicy.php

```diff
@@ -15,8 +15,30 @@ public function edit(User $user, UserPost $post): bool
         return $user->id === $post->user_id;
     }
 
+    /**
+     * Check if the user can delete a post.
+     */
     public function delete(User $user, UserPost $post): bool
     {
         return $user->id === $post->user_id;
     }
+
+    /**
+     * Check if the user can view a post.
+     */
+    public function view(User $user, UserPost $post): bool
+    {
+        if ($user != null) {
+
+            $isFollowing = ($user->isFollowing($post->user));
+            if ($user != null && $post->user != null) {
+                $isOwner = $user->id === $post->user->id;
+            } else {
+                $isOwner = false;
+            }
+            return  $isFollowing || $isOwner;
+        }
+
+        return false;
```

Comment:
Prevent using else in if statement.

Reply:
done sir, apologies for overlooking this. Thank you